src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -128,7 +128,7 @@ static string ValueToString(object value)
             if (list1 != null)
             {
                 var list = list1;
-                return String.Join(";", list.Cast<object>().Select(ValueToString));
+                return String.Join(";", list.Cast<object>().Select(o => ValueToString(o)));
```

Comment:
The advantage here is that since this is a lambda that doesn't close over anything, the compiler generates code that creates and caches the delegate once and reuses it for every call.

This optimization doesn't happen you pass in a method group, so every call allocates a new delegate. There is an open PR on the Roslyn repo to change this behavior.